osbs/core.py

```diff
@@ -405,20 +404,12 @@ def stream_logs(self, build_id):
         while True:
             buildlogs_url = self._build_url("builds/%s/log/" % build_id,
                                             **kwargs)
-            try:
-                response = self._get(buildlogs_url, stream=1,
-                                     headers={'Connection': 'close'})
-            except OsbsNetworkException as ex:
-                # pycurl reports 'empty reply from server' as
-                # FOLLOWLOCATION. Handle this as though there were no
-                # lines returned.
-                if ex.status_code != pycurl.FOLLOWLOCATION:
-                    raise
-            else:
-                check_response(response)
-                for line in response.iter_lines():
-                    last_activity = time.time()
-                    yield line
+            response = self._get(buildlogs_url, stream=1,
+                                 headers={'Connection': 'close'})
```

Comment:
I just tested this. `requests` *does* treat this as an error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/site-packages/requests/api.py", line 71, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/api.py", line 57, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 475, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 617, in send
    r.content
  File "/usr/lib/python2.7/site-packages/requests/models.py", line 741, in content
    self._content = bytes().join(self.iter_content(CONTENT_CHUNK_SIZE)) or bytes()
  File "/usr/lib/python2.7/site-packages/requests/models.py", line 667, in generate
    raise ChunkedEncodingError(e)
requests.exceptions.ChunkedEncodingError: ('Connection broken: IncompleteRead(0 bytes read)', IncompleteRead(0 bytes read))
```

You can test it yourself using this:
```
from http.server import HTTPServer, BaseHTTPRequestHandler

class ChunkServer(BaseHTTPRequestHandler):
    def do_GET(self):
        self.send_response(200)
        self.send_header('Transfer-Encoding', 'chunked')
        self.end_headers()

chunk = HTTPServer(('', 5700), ChunkServer)
chunk.serve_forever()
```

We need to make sure this case is handled correctly and that osbs-client will re-connect in this situation.